server/mailet/integration-testing/src/test/java/org/apache/james/mailets/NetworkMatcherIntegrationTest.java

```diff
@@ -0,0 +1,323 @@
+                .matcherCondition("126.0.0.0/4")
+                .mailet(ToProcessor.class)
+                .addProperty("processor", "transport")
+                .build())
+            .addMailet(dropAllMailMailet()));
+
+        try (SMTPMessageSender messageSender =
+                 SMTPMessageSender.authentication(LOCALHOST_IP, SMTP_PORT, JAMES_APACHE_ORG, FROM, PASSWORD);
+             IMAPMessageReader imapMessageReader = new IMAPMessageReader(LOCALHOST_IP, IMAP_PORT)) {
+
+            messageSender.sendMessage(FROM, FROM);
+            calmlyAwait.atMost(Duration.ONE_MINUTE).until(messageSender::messageHasBeenSent);
+
+            calmlyAwait.atMost(Duration.ONE_MINUTE)
+                .until(() -> imapMessageReader.userReceivedMessage(FROM, PASSWORD));
+        }
+    }
+
+    @Test
+    public void mailsFromNonAuthorizedNetworksShouldNotBeDeliveredWithRemoteAddrInNetwork() throws Exception {
+        jamesServer = createJamesServerWithRootProcessor(ProcessorConfiguration.builder()
+            .state("root")
+            .addMailet(MailetConfiguration.builder()
+                .matcher(RemoteAddrInNetwork.class)
+                .matcherCondition("172.0.0.0/8")
+                .mailet(ToProcessor.class)
+                .addProperty("processor", "transport")
+                .build())
+            .addMailet(dropAllMailMailet()));
+
+        try (SMTPMessageSender messageSender =
+                 SMTPMessageSender.noAuthentication(LOCALHOST_IP, SMTP_PORT, JAMES_APACHE_ORG);
+             IMAPMessageReader imapMessageReader = new IMAPMessageReader(LOCALHOST_IP, IMAP_PORT)) {
+
+            messageSender.sendMessage(FROM, FROM);
+
+            calmlyAwait.atMost(Duration.ONE_MINUTE).until(messageSender::messageHasBeenSent);
+
+            Thread.sleep(TimeUnit.SECONDS.toMillis(10));
+
+            assertThat(imapMessageReader.userReceivedMessage(FROM, PASSWORD)).isFalse();
+        }
+    }
+
+    @Test
+    public void mailsFromNonAuthorizedNetworksShouldNotBeDeliveredWithRemoteAddrNotInNetwork() throws Exception {
+        jamesServer = createJamesServerWithRootProcessor(ProcessorConfiguration.builder()
+            .state("root")
+            .addMailet(MailetConfiguration.builder()
+                .matcher(RemoteAddrNotInNetwork.class)
+                .matcherCondition("127.0.0.0/8")
+                .mailet(ToProcessor.class)
+                .addProperty("processor", "transport")
+                .build())
+            .addMailet(dropAllMailMailet()));
+
+        try (SMTPMessageSender messageSender =
+                 SMTPMessageSender.noAuthentication(LOCALHOST_IP, SMTP_PORT, JAMES_APACHE_ORG);
+             IMAPMessageReader imapMessageReader = new IMAPMessageReader(LOCALHOST_IP, IMAP_PORT)) {
+
+            messageSender.sendMessage(FROM, FROM);
+
+            calmlyAwait.atMost(Duration.ONE_MINUTE).until(messageSender::messageHasBeenSent);
+
+            Thread.sleep(TimeUnit.SECONDS.toMillis(10));
```

Comment:
what is that ?

Reply:
How do you ensure something do not happen?

You have a better way doing than **checking after a fixed amount of time**? Because if you do I take it! That would be quite of a nice theorical revolution.

Reply:
VacationMailet is not needed but bcc stripping is.